stacks/container-apps/sphinx-nlb.yml

```diff
@@ -68,161 +68,187 @@ Parameters:
-  #     ComparisonOperator: GreaterThanThreshold
-  #     EvaluationPeriods: 3
-  #     MetricName: HTTPCode_ELB_5XX_Count
-  #     Namespace: AWS/ApplicationELB
-  #     Period: 60
-  #     Statistic: Sum
-  #     Threshold: 2
-  #     TreatMissingData: notBreaching
-  #     Dimensions:
-  #       - Name: LoadBalancer
-  #         Value: !GetAtt NLB.LoadBalancerFullName
-  # NLBResponseTimeP95Alarm:
-  #   Type: "AWS::CloudWatch::Alarm"
-  #   Condition: CreateProductionResources
-  #   Properties:
-  #     ActionsEnabled: true
-  #     AlarmName: !Sub "[${AppName}][NLB][TargetResponseTime] P95 extremely slow"
-  #     AlarmActions:
-  #       - !Ref OpsWarnMessagesSnsTopicArn
-  #     InsufficientDataActions:
-  #       - !Ref OpsWarnMessagesSnsTopicArn
-  #     OKActions:
-  #       - !Ref OpsWarnMessagesSnsTopicArn
-  #     AlarmDescription: !Sub |
-  #       Target response time value for ${AppName} load balancer targets at 95th
-  #       percentile higher than expected
-  #     ComparisonOperator: GreaterThanOrEqualToThreshold
-  #     EvaluationPeriods: 1
-  #     ExtendedStatistic: p95
-  #     MetricName: TargetResponseTime
-  #     Namespace: AWS/ApplicationELB
-  #     Period: 300
-  #     Threshold: 0.2
-  #     TreatMissingData: notBreaching
-  #     Dimensions:
-  #       - Name: LoadBalancer
-  #         Value: !GetAtt NLB.LoadBalancerFullName
-  # # ECS Service - Web
+  NLB:
+    Type: "AWS::ElasticLoadBalancingV2::LoadBalancer"
+    Properties:
+      IpAddressType: ipv4
+      Scheme: internet-facing
+      Subnets:
+        - !Ref VPCSubnet1
+        - !Ref VPCSubnet2
+        - !Ref VPCSubnet3
+      Type: network
+      Tags:
+        - Key: Project
+          Value: !Ref AppName
+        - Key: Environment
+          Value: !Ref EnvironmentType
+        - Key: Name
+          Value: !Sub ${AppName}-${EnvironmentType}
+        - Key: "prx:cloudformation:stack-name"
+          Value: !Ref AWS::StackName
+        - Key: "prx:cloudformation:stack-id"
+          Value: !Ref AWS::StackId
+  NLBSphinxListener:
+    Type: "AWS::ElasticLoadBalancingV2::Listener"
+    Properties:
+      LoadBalancerArn: !Ref NLB
+      Port: 9312
+      Protocol: TCP
+      DefaultActions:
+        - Type: forward
+          TargetGroupArn: !Ref NLBTargetGroup
+  NLBHTTPListener:
+    Type: "AWS::ElasticLoadBalancingV2::Listener"
+    Properties:
+      LoadBalancerArn: !Ref NLB
+      Port: 80
+      Protocol: TCP
+      DefaultActions:
+        - Type: forward
+          TargetGroupArn: !Ref NLBTargetGroup
+  NLBHTTPSListener:
+    Type: "AWS::ElasticLoadBalancingV2::Listener"
+    Properties:
+      Certificates:
+        - CertificateArn: !Ref VPCCertificateArn
+      LoadBalancerArn: !Ref NLB
+      Port: 443
+      Protocol: TLS
+      SslPolicy: "ELBSecurityPolicy-2016-08"
+      DefaultActions:
+        - Type: forward
+          TargetGroupArn: !Ref NLBTargetGroup
+  NLBTargetGroup:
```

Comment:
Maybe this could be called `NLBWebTargetGroup` or something more distinct

Reply:
yeah, and I think I have a mistake in using these with the listeners...